File: pyg-manager-web/src/main/webapp/plugins/adminLTE/js/controller/brandController.js
app.controller('brandController', function ($scope,$controller, brandService) {
    //控制器继承
    //继承baseController控制器，且把父控制器$scope传递子控制器的$scope
    $controller("baseController",{$scope:$scope});
    //定义函数
    //向服务器发送ajax请求
    $scope.findAll = function () {
        brandService.findAll().success(function (data) {
            $scope.list = data;
        });
    };


    $scope.findPage = function (pageNum, pageSize) {
        brandService.findPage(pageNum, pageSize).success(function (data) {
            $scope.list = data.rows;
            $scope.paginationConf.totalItems = data.total;
        })
    };
    $scope.save = function () {
        var objService = null;
        if ($scope.entity.id != null) {
            objService = brandService.updata($scope.entity);
        } else {
            objService = brandService.save($scope.entity);
        }

        objService.success(function (data) {
            if (data.success) {
                $scope.reloadList();
            } else {
                alert("保存失败")
            }
        })
    };
    $scope.findById = function (id) {
        brandService.findById(id).success(function (data) {
            $scope.entity = data;
        })
    };

    $scope.del = function () {
        brandService.del($scope.delList).success(function (data) {
            if (data.success) {
                $scope.delList = [];
                $scope.reloadList();
            } else {
                alert("删除失败")
            }
        })
    }

});

File: pyg-shop-web/src/main/java/com/pyg/shop/controller/BrandController.java
package com.pyg.shop.controller;


import com.alibaba.dubbo.config.annotation.Reference;
import com.pyg.maneger.service.BrandService;
import com.pyg.pojo.TbBrand;
import com.pyg.utils.PageResult;
import com.pyg.utils.pygResult;

import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;


@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;
    @RequestMapping("findAll")
    public List<TbBrand> findAll(){
        List<TbBrand> tbBrandList = brandService.findAll();
        return tbBrandList;
    }
    @RequestMapping("findPage/{pageNum}/{pageSize}")
    public PageResult findPage(@PathVariable int pageNum,@PathVariable int pageSize ){
        //调用brandService
        PageResult page = brandService.findPage(pageNum, pageSize);
        return page;
    }
    @RequestMapping("/save")
    public pygResult save(@RequestBody TbBrand tbBrand){
        try {
            brandService.save(tbBrand);
           return new pygResult(true,"保存成功");
        } catch (Exception e) {
            e.printStackTrace();
           return new pygResult(false,"保存失败");

        }
    }
    @RequestMapping("/findById/{id}")
    public TbBrand findById(@PathVariable Long id){
        TbBrand tbBrand = brandService.findById(id);
        return tbBrand;
    }
    @RequestMapping("/updata")
    public pygResult updata(@RequestBody TbBrand tbBrand){
        try {
            brandService.updata(tbBrand);
            return new pygResult(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new pygResult(false,"保存失败");
        }
    }
    @RequestMapping("/del/{ids}")
    public pygResult del(@PathVariable Long [] ids){
        try {
            brandService.del(ids);
            return new pygResult(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
           return new pygResult(false,"删除失败");
        }
    }
    @RequestMapping("findbrandList")
    public List<Map> findbrandList(){
      return   brandService.findbrandList();
    }
}
